app/pages/challenges/components/challengeForm.tsx

```diff
@@ -143,37 +222,72 @@ export const ChallengeForm: React.FC<ChallengeFormProps> = ({
               {errors.unit && <p className="text-red-500">{errors.unit}</p>}
             </div>
           </div>
-          <div className="space-y-2">
-            <Label htmlFor="friendIds">Invite Friends (Enter Friend ID)</Label>
-            <div className="flex space-x-2">
+          <div className="space-y-2" ref={searchRef}>
+            <Label>Invite Friends</Label>
+            <div className="relative">
               <Input
-                id="friendIds"
-                name="friendIds"
-                value={friendIds}
-                onChange={(e) => setFriendIds(e.target.value)}
+                placeholder="Search for friends..."
+                value={searchTerm}
+                onChange={(e) => setSearchTerm(e.target.value)}
+                onFocus={() => setIsSearchOpen(true)}
+                onClick={() => setIsSearchOpen(true)}
+                disabled={isLoading}
               />
-              <Button type="button" onClick={handleAddFriend}>
-                Add
-              </Button>
+
+              {isLoading && (
+                <div className="absolute right-3 top-1/2 transform -translate-y-1/2">
+                  <div className="h-4 w-4 border-2 border-primary border-t-transparent rounded-full animate-spin"></div>
+                </div>
+              )}
+
+              {isSearchOpen && (
+                <div className="absolute z-10 w-full mt-1 bg-background border rounded-md shadow-lg max-h-60 overflow-auto">
+                  {loadError ? (
+                    <div className="p-2 text-sm text-red-500">{loadError}</div>
+                  ) : filteredFriends.length === 0 ? (
+                    <div className="p-2 text-sm text-muted-foreground">
+                      {isLoading ? "Loading friends..." : "No friends found"}
+                    </div>
+                  ) : (
+                    <ul>
+                      {filteredFriends.map((friend) => (
+                        <li
+                          key={friend.id}
+                          className="px-3 py-2 cursor-pointer hover:bg-secondary/50"
+                          onClick={() => handleAddFriend(friend)}
+                        >
+                          {friend.name} (ID: {friend.id})
```

Comment:
pleae extract logic into seperate component. we need to reuse this input multiple times in several places , not only in challengeForm